### PR TITLE
Autom/AC BSA 3 BSA 1

### PR DESCRIPTION
**Pull request by AlanDiCha123**
# This branch includes all the automated tesings for all fields with its expecting data
I also would include the automation with Jenkins, but unfortunately it's turned off :(

### CODE REVIEW

File: Pages/edit_customer_page.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+from selenium.webdriver.chrome.webdriver import WebDriver
+from Base.page_base import PageBase
+from Locators.locators import EditCustomerLocators, EditCustomerLoginLocators
+
+
+class EditCustomerPage(PageBase):
+    def __init__(self, driver: WebDriver):
+        super().__init__(driver)
+    
+    def click_on_login(self):
```
Comment (eduxsan) [inline Pages/edit_customer_page.py]: Please change this name, actual is a little bit misleading

File: Pages/edit_customer_page.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    
+    def verify_data_by_field(self, input_name, expected_data):
+        try:
+            element = ""
+            current_data = ""
+            if input_name == "Customer Name":
+                element = self.driver.find_element(*EditCustomerLocators.customer_name_value)
+            elif input_name == "PIN":
+                element = self.driver.find_element(*EditCustomerLocators.pin_value)
+
```
Comment (eduxsan) [inline Pages/edit_customer_page.py]: Missing rest of the fields: Address, City, State, Telephone, Gender, ...

File: Pages/edit_customer_page.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            if input_name == "Customer Name":
+                element = self.driver.find_element(*EditCustomerLocators.customer_name_value)
+            elif input_name == "PIN":
+                element = self.driver.find_element(*EditCustomerLocators.pin_value)
+            
+            current_data = element.get_attribute("value")
+        except Exception as e:
+            print(e)
+
+        assert current_data == expected_data, f'El mensaje en el campo {input_name} es {current_data}. Se esperaba {expected_data}'
```
Comment (eduxsan) [inline Pages/edit_customer_page.py]: Please update error message

File: Pages/new_customer_page.py [hunk 4]
```diff
@@ -33,3 +62,71 @@ def verify_message_by_input_name(self, input_name, expected_message):
...
+            self.driver.find_element(*NewCustomerLocators.phone_name_txt).send_keys(phone)
+        except Exception as e:
+            print(e)
+
+    def write_email(self) -> None:
+
+        length = 5
+        letters = string.ascii_lowercase
+        result_str = ''.join(random.choice(letters) for _ in range(length))
+        result_str += '@aol.com'
```
Comment (eduxsan) [inline Pages/new_customer_page.py:128]: You can return random email or move this logic on other method, something that allows us to know what is the email we provided for the costumer

File: Tests/New_Customer/test_new_customer.py [hunk 5]
```diff
@@ -20,6 +21,51 @@ def test_new_customer_input_message(self):
...
+
+        common.click_on_button_menu("Edit Customer")
+        time.sleep(2)
+
+        editCustomer = EditCustomerPage(self.driver)
+        editCustomer.write_id(id_data)
+        editCustomer.click_on_login()
+        time.sleep(5)
+        editCustomer.verify_data_by_field("PIN", "548320")
+        editCustomer.verify_data_by_field("Customer Name", "Holmes")
```
Comment (eduxsan) [inline Tests/New_Customer/test_new_customer.py]: Missing validations, as mentioned on previous comment

File: Tests/New_Customer/test_new_customer.py [hunk 6]
```diff
@@ -20,6 +21,51 @@ def test_new_customer_input_message(self):
...
+        login.write_username("mngr547360")
+        login.write_password("YjAnEtU")
+        login.click_on_login()
+        time.sleep(2)
+        common = CommonPage(self.driver)
+        common.click_on_button_menu("New Customer")
+
+        newCustomer = NewCustomerPage(self.driver)
+        newCustomer.write_customer_name("Holmes")
+        newCustomer.write_date_of_birth("2072000")
```
Comment (eduxsan) [inline Tests/New_Customer/test_new_customer.py]: Missing select gender

**Review by eduxsan — CHANGES_REQUESTED** [review]

**Review by eduxsan — APPROVED** [review]

**Review by LanderoX144 — APPROVED** [review]

**Pull request merged**